### PR TITLE
add statistics record function

### PR DESCRIPTION
**Pull request by mengzhuwang**
please check it and draw a new panel for it on the mainframe
@liuhaipei

### CODE REVIEW

**Comment by mengzhuwang** [comment]
we finished this function, please check @iroq

**Comment by iroq** [comment]
It doesn't seem to work...
* With the simulation, at the start it's fluctuating quite wildly, it can go from 0.013 to 0.006 which is very big. The vehicles move at a constant speed so the average speed should be equal to their constant speed (give or take rounding errors). See [video](http://tinypic.com/r/25hnvab/9)
* When I click Clear, it's getting even crazier. See [video](http://tinypic.com/r/zlrlug/9)
* Starting with `FirstVersionStateProvider` shows a clear bug. Average speed is constantly falling, then jumps to 0 when it switches the roads, and then slowly grows. It should be constant here. See [video](http://tinypic.com/r/120nfxi/9)
* Starting with `JunctionLocationTestProvider` crashes instantly (throws `NullPointerException`).

File: src/main/java/thewaypointers/trafficsimulator/gui/MapPanel.java [hunk 1]
```diff
@@ -70,8 +84,63 @@ public void NewStateReceived(WorldStateDTO worldStateDTO){
...
+        double normalCarTotalSpeed = 0;
+        int normalCarTotalNum = GetALLStatistics(VehicleType.CarNormal).size();
+        for (Statistics s : GetALLStatistics(VehicleType.CarNormal)){
+
+            //normalCarTotalNum = normalCarTotalNum++;
+            if(s.getTime()!=0) {
+                normalCarTotalSpeed = normalCarTotalSpeed + s.getDistance() / s.getTime();
+                System.out.println("normal car total speed: " + normalCarTotalSpeed);
+
+                System.out.println("normal car total num: " + normalCarTotalNum);
```
Comment (iroq) [inline src/main/java/thewaypointers/trafficsimulator/gui/MapPanel.java]: Can you remove the prints? It's a lot on the console.

**Comment by mengzhuwang** [comment]
1.for the problem one, fixed.
2.clear button does not work in my computer. I cannot check.
3.In FirstVersionStateProvider, the car jumps at the junction, that why it becomes 0 afterwards. we assume distance start with roads and one road must after a junction. In JunctionLocationTestProvider is the same reason.

**Comment by iroq** [comment]
Clear button works, you just have to start the simulation again for the state to clean (because we as of now don't have a "clean" state ready that we could display immidiately after clicking Clear).

Ok, you can assume that, but you should fix `JunctionLocationTestStarter` so it's possible to use it again. Also, looking into why the speed decreases in `FirstVersionStarter` while the vehicle is still on the first road might help with debugging, because from GUI's point of view it's no different than in the real simulation, but it's bugged.

**Comment by iroq** [comment]
Nice.

**Pull request merged**